pulsar-common/src/main/java/org/apache/pulsar/common/api/Commands.java

```diff
@@ -282,10 +282,11 @@ public static MessageMetadata parseMessageMetadata(ByteBuf buffer) {
         }
     }
 
-    public static ByteBufPair newMessage(long consumerId, MessageIdData messageId, ByteBuf metadataAndPayload) {
+    public static ByteBufPair newMessage(long consumerId, MessageIdData messageId, int redeliveryCount, ByteBuf metadataAndPayload) {
         CommandMessage.Builder msgBuilder = CommandMessage.newBuilder();
         msgBuilder.setConsumerId(consumerId);
         msgBuilder.setMessageId(messageId);
+        msgBuilder.setRedeliveryCount(redeliveryCount);
```

Comment:
if the `redeliveryCount` is 0, which is the default in the proto definition, and it's passed when the feature is disabled, we should avoid setting the value on the protobuf builder. If that is omitted, it would save 2 bytes per each message.